compile error on mac with develop  branch code

/Users/xingzhaolong/hg/paddle_project/paddle_me/paddle/function/ConvOp.h:76:8: error: 'paddle::ConvFunctionBase::check' hides overloaded virtual function
      [-Werror,-Woverloaded-virtual]
  void check(const TensorShape& input,
       ^
/Users/xingzhaolong/hg/paddle_project/paddle_me/paddle/function/Function.h:177:16: note: hidden overloaded virtual function 'paddle::FunctionBase::check'
      declared here: different number of parameters (2 vs 3)
  virtual void check(const BufferArgs& inputs, const BufferArgs& outputs) {}
               ^
In file included from /Users/xingzhaolong/hg/paddle_project/paddle_me/paddle/function/GemmConvOp.cpp:15:
In file included from /Users/xingzhaolong/hg/paddle_project/paddle_me/paddle/function/GemmConvOp.h:17:
/Users/xingzhaolong/hg/paddle_project/paddle_me/paddle/function/ConvOp.h:71:16: error: 'calc' overrides a member function but is not marked 'override'
      [-Werror,-Winconsistent-missing-override]
  virtual void calc(const BufferArgs& inputs, const BufferArgs& outputs) {}
               ^
/Users/xingzhaolong/hg/paddle_project/paddle_me/paddle/function/Function.h:171:16: note: overridden virtual function is here
  virtual void calc(const BufferArgs& inputs, const BufferArgs& outputs) {}
               ^
2 errors generated.
make[2]: *** [paddle/function/CMakeFiles/paddle_function.dir/GemmConvOp.cpp.o] Error 1
make[1]: *** [paddle/function/CMakeFiles/paddle_function.dir/all] Error 2
make: *** [all] Error 2

--- a/paddle/function/ConvOp.h
+++ b/paddle/function/ConvOp.h
@@ -68,14 +68,12 @@ public:
     numOutputs_ = 1;
   }
 
-  virtual void calc(const BufferArgs& inputs, const BufferArgs& outputs) {}
-
   // input can be INPUT and INPUT_GRAD
   // filter can be FILTER and FILTER_GRAD
   // output can be OUTPUT and OUTPUT_GRAD
-  void check(const TensorShape& input,
-             const TensorShape& filter,
-             const TensorShape& output) {
+  void checkShape(const TensorShape& input,
+                  const TensorShape& filter,
+                  const TensorShape& output) {
     // inputs and outputs arguments should be 4-dimensional.
     CHECK_EQ(input.ndims(), (size_t)4);
     CHECK_EQ(output.ndims(), (size_t)4);

--- a/paddle/function/GemmConvOp.cpp
+++ b/paddle/function/GemmConvOp.cpp
@@ -117,15 +117,23 @@ public:
     ConvFunctionBase::init(config);
   }
 
+  virtual void check(const BufferArgs& inputs,
+                     const BufferArgs& outputs) override {
+    const TensorShape& input = inputs[0].shape();
+    const TensorShape& filter = inputs[1].shape();
+    const TensorShape& output = outputs[0].shape();
+    checkShape(input, filter, output);
+  }
+
   void calc(const BufferArgs& inputs, const BufferArgs& outputs) override {
     CHECK_EQ(numInputs_, inputs.size());
     CHECK_EQ(numOutputs_, outputs.size());
+    check(inputs, outputs);
     // TODO(hedaoyuan): Need to define some index macros,
     // to avoid useing 0 and 1.
     const TensorShape& input = inputs[0].shape();
     const TensorShape& filter = inputs[1].shape();
     const TensorShape& output = outputs[0].shape();
-    check(input, filter, output);
 
     real beta;
     if (outputs[0].getArgType() == ADD_TO) {
@@ -209,16 +217,24 @@ public:
     ConvFunctionBase::init(config);
   }
 
+  virtual void check(const BufferArgs& inputs,
+                     const BufferArgs& outputs) override {
+    const TensorShape& output = inputs[0].shape();
+    const TensorShape& filter = inputs[1].shape();
+    const TensorShape& input = outputs[0].shape();
+    checkShape(input, filter, output);
+  }
+
   void calc(const BufferArgs& inputs, const BufferArgs& outputs) override {
     CHECK_EQ(numInputs_, inputs.size());
     CHECK_EQ(numOutputs_, outputs.size());
+    check(inputs, outputs);
     // Since the implementation of Col2ImFunctor is ADD_TO,
     // this function only supports ADD_TO mode.
     CHECK_EQ(outputs[0].getArgType(), ADD_TO);
     const TensorShape& output = inputs[0].shape();
     const TensorShape& filter = inputs[1].shape();
     const TensorShape& input = outputs[0].shape();
-    check(input, filter, output);
 
     size_t batchSize = input[0];
     size_t inputChannels = input[1];
@@ -295,13 +311,21 @@ public:
     ConvFunctionBase::init(config);
   }
 
-  void calc(const BufferArgs& inputs, const BufferArgs& outputs) override {
-    CHECK_EQ(numInputs_, inputs.size());
-    CHECK_EQ(numOutputs_, outputs.size());
+  virtual void check(const BufferArgs& inputs,
+                     const BufferArgs& outputs) override {
     const TensorShape& output = inputs[0].shape();
     const TensorShape& input = inputs[1].shape();
     const TensorShape& filter = outputs[0].shape();
-    check(input, filter, output);
+    checkShape(input, filter, output);
+  }
+
+  void calc(const BufferArgs& inputs, const BufferArgs& outputs) override {
+    CHECK_EQ(numInputs_, inputs.size());
+    CHECK_EQ(numOutputs_, outputs.size());
+    check(inputs, outputs);
+    const TensorShape& output = inputs[0].shape();
+    const TensorShape& input = inputs[1].shape();
+    const TensorShape& filter = outputs[0].shape();
 
     real beta;
     if (outputs[0].getArgType() == ADD_TO) {

--- a/paddle/function/NaiveConvOp.cpp
+++ b/paddle/function/NaiveConvOp.cpp
@@ -90,14 +90,19 @@ public:
     ConvFunctionBase::init(config);
   }
 
-  void calc(const BufferArgs& inputs, const BufferArgs& outputs) override {
-    CHECK_EQ(numInputs_, inputs.size());
-    CHECK_EQ(numOutputs_, outputs.size());
+  virtual void check(const BufferArgs& inputs,
+                     const BufferArgs& outputs) override {
     const TensorShape& input = inputs[0].shape();
     const TensorShape& filter = inputs[1].shape();
     const TensorShape& output = outputs[0].shape();
-    check(input, filter, output);
+    checkShape(input, filter, output);
+  }
+
+  void calc(const BufferArgs& inputs, const BufferArgs& outputs) override {
+    CHECK_EQ(numInputs_, inputs.size());
+    CHECK_EQ(numOutputs_, outputs.size());
     CHECK_EQ(outputs[0].getArgType(), ASSIGN_TO);
+    check(inputs, outputs);
 
     size_t batchSize = inputs[0].shape()[0];
     size_t inputChannels = inputs[0].shape()[1];

